src/main/java/com/spring/onlinebookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,20 @@
+package com.spring.onlinebookstore.security;
+
+import com.spring.onlinebookstore.repository.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException(username));
```

Comment:
provide informative message

```suggestion
                .orElseThrow(() -> new EntityNotFoundException(username));
```